app/src/main/java/com/ilatyphi95/farmersmarket/utils/NetworkAvailabilityUtils.kt

```diff
@@ -0,0 +1,47 @@
+package com.ilatyphi95.farmersmarket.utils
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkRequest
+import android.os.Build
+
+
+object NetworkAvailabilityUtils {
+
+    private lateinit var networkAvailabilityListener: (Boolean) -> Unit
+
+    private fun isNetworkAvailable(context: Context) {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            val builder = NetworkRequest.Builder()
+            connectivityManager.registerNetworkCallback(builder.build(),
+                object : ConnectivityManager.NetworkCallback() {
+                    override fun onAvailable(network: Network) {
+                        super.onAvailable(network)
+                        networkAvailabilityListener(true)
+                    }
+
+                    override fun onLost(network: Network) {
+                        super.onLost(network)
+                        networkAvailabilityListener(false)
+                    }
+                })
+
+
+        } else {
+            val networkInfo = connectivityManager.activeNetworkInfo
+            networkAvailabilityListener(networkInfo != null && networkInfo.isConnected)
```

Comment:
This make use of deprecated code, can you find non deprecated alternatives?

Reply:
Yes. I didn't find a better way to do it